src/metax_api/api/rest/base/views/file_view.py

```diff
@@ -174,8 +174,16 @@ def datasets(self, request):
         requests to query parameters, so using POST instead is more guaranteed to work.
         """
 
+        keysonly = CommonService.get_boolean_query_param(self.request, 'keysonly')
+
         if CommonService.get_boolean_query_param(request, 'detailed'):
-            return FileService.get_detailed_datasets_where_file_belongs_to(request.data)
+            return FileService.get_detailed_datasets_where_file_belongs_to(request.data, keysonly=False)
+
+        if 'keys' in request.query_params:
+            if 'datasets' in request.query_params['keys']:
```

Comment:
value of 'keys' is expected to be a string so equal sign should be used here. Now for example 'FOOdatasetsFOO' is true here also

Reply:
x/ you're right, will fix it...